src/taskManager/UserInterface.java

```diff
@@ -0,0 +1,114 @@
+package taskManager;
+
+import java.util.Scanner;
+
+public class UserInterface {
+    Scanner sc = new Scanner(System.in);
+    TaskList taskList = new TaskList();
```

Comment:
This UserInterface class acts like the whole application cause it's not responsible only for UI - it also creates and maintains the application "model" (task list).

One of the easy improvements could be adding a constructor that accepts TaskList, so that UI at least does not create the model.

Another improvement could be removing the knowledge about TaskList and Task and using the Observer/EventListener pattern. In this case, there would also be a "controller" created that would connect "ui" with "model". This is called an MVC pattern (model view controller).

Reply:
It seems I have a lot of work to do, but for now, I'll stick with your easy improvement suggestion. :)